.github/workflows/cd.yml

```diff
@@ -453,15 +455,16 @@ jobs:
             }
 
             // Parse and filter environments
-            const allowedEnvironments = ['dev', 'ops', 'prod'];
+            const allowedEnvironments = ['dev', 'ops', 'prod-canary', 'prod'];
             let environments = [];
             environments = ENVIRONMENT.split(',')
               .map(e => e.trim())
               .filter(e => allowedEnvironments.includes(e));
 
             // Special case: 'prod' means we deploy to all environments
             if (environments.includes('prod')) {
-              environments = allowedEnvironments;
+              // no need to deploy to prod-canary because it is a subset of prod
+              environments = allowedEnvironments.filter(e => e !== 'prod-canary');
```

Comment:
When we have prod → we are publishing to all the catalogs, no need to publish twice for prod. Argo workflow will take care of prod-canary trigger in the [code](https://github.com/grafana/deployment_tools/blob/master/ksonnet/environments/grafana-plugins-cd/main.jsonnet#L41)